torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -481,10 +482,20 @@ ProcessGroupNCCL::WorkNCCL::WorkNCCL(
   // Note: The actual events are lazily created when first recorded to with
   // DEFAULT_FLAGS = cudaEventDisableTiming.
   if (enableTiming) {
-    ncclStartEvent_ = std::make_shared<at::cuda::CUDAEvent>(cudaEventDefault);
+    if (cudaEventCacheEnabled) {
+      ncclStartEvent_ =
+          ProcessGroupNCCL::CUDAEventCache::get().create(enableTiming);
+    } else {
+      ncclStartEvent_ = std::make_shared<at::cuda::CUDAEvent>(cudaEventDefault);
+    }
+  }
+  if (cudaEventCacheEnabled) {
```

Comment:
Is that possible to make the 4 execution branches clearer using the nested if/else?